### DISCUSSION TITLE
Confusion about error.tsx vs global-error.tsx

### DISCUSSION THREAD

**Discussion by g1na1011** [vercel/next.js#68048] [App Router]
Hi there! We are using Next.js App Router to develop a new architecture for our app. We are starting off with a pure static page that is fairly simple, no nested routes required. Can you help me explain why or when we would want both an `error.tsx` and a `global-error.tsx` page? Our error pages would have the same design, so we thought we can remove `error.tsx` and just keep `global-error.tsx`. To our surprise, using just `global-error.tsx` renders a blank white page when we throw a 500. Renaming that page back to `error.tsx` renders the expected page. Should we have both even though the designs are the same and we do not have a component-based error page? 

thank you for your help!

**Comment by knarrff** [comment]
Hi @g1na1011 

From what I read from [here](https://nextjs.org/docs/app/api-reference/file-conventions/error) and [here](https://nextjs.org/docs/app/building-your-application/routing/error-handling), you need both:

"_An error.js boundary will not handle errors thrown in a layout.js component in the same segment because the error boundary is nested inside that layouts component._

- _To handle errors for a specific layout, place an error.js file in the layouts parent segment._
- _To handle errors within the root layout or template, use a variation of error.js called app/global-error.js._ "

"_The root app/error.js boundary does not catch errors thrown in the root app/layout.js or app/template.js component._

_To specifically handle errors in these root components, use a variation of error.js called app/global-error.js located in the root app directory._"

Which I understand as: global-error.js isn't used when error.js should be but isn't there.

**Reply by g1na1011** [reply]
@knarrff Thanks for your response and clarifying! I've read that previously, but thought that the since this app currently only has one route, `app/layout.tsx`, we can utilize `global-error.tsx` to catch the error instead of having both `global-error.tsx` and `error.tsx` under the `app/` directory. I guess my confusion started by not fully understanding what "segments" mean.

In this case, do you think it still makes sense to have two error files with the same design? Thank you for your help!

**Comment by ksuraj112** [comment]
@g1na1011 The issue is you might not be adding <html> and <body> tags in the `global-error.tsx` file, that is why you were seeing blank white page. `global-error.tsx` is capable of handling all errors since its on top. `error.tsx` is just to prevent the layout styles from crashing while showing your custom fallback error page.
